R/prep_sedt_resource.R

```diff
@@ -0,0 +1,135 @@
+#' Prepare a resource for upload to the Spatial Equity Data Tool API
+#'
+#' [prep_sedt_resource()] is used by [call_sedt_api()] to convert an object or
+#' URL to a valid file path before uploading the file to the Spatial Equity Data
+#' Tool API.
+#'
+#' @param resource A URL, file path, sf or sfc object, or a data frame with
+#'   `coords` columns. See details for more information.
+#' @param coords A length 2 character vector with coordinate column names in
+#'   lon/lat order. Should correspond to the `resource_lon_column` and
+#'   `resource_lat_column` values for [call_sedt_api()].
+#' @param file File path for output file. Optional. If `NULL`, file path is set
+#'   to a temporary directory.
+#' @param placement Placement of point when converting non-point input geometry.
+#'   "surface" (default) or "centroid". Only used if resource is a file path for
+#'   a spatial data file or an `sf` object. See details for more information.
+#' @inheritParams rlang::args_error_context
+#' @details Valid resource inputs
+#'
+#' To call the Spatial Equity Data Tool API, [call_sedt_api()] requires a CSV or
+#' TSV input file with coordinate attribute columns. [prep_sedt_resource()] also allows users to alternatively supply:
+#'
+#' - A URL for a delimited file with coordinate columns or a spatial data file
+#' - A ArcGIS FeatureLayer or Table URL (read with [arcgislayers::arc_read()])
+#' - A file path for a spatial or tabular data file that can be read with [sf::read_sf()]
+#' - A data frame containing the named columns in `coords`
+#' - A `sf` or `sfc` object that can be converted to a data frame using [sf::st_coordinates()]
+#'
+#' If resource is an `sf` object or a path to a spatial data file with POLYGON,
+#' MULTIPOLYGON, or any non-POINT geometry, `placement` is used to convert the
+#' input into points. [sf::st_centroid()] (if `placement = "centroid"`) or
+#' [sf::st_point_on_surface()] (if `placement = "surface"` which is the default)
+#' option.
+#' @returns A path to a CSV or TSV file (depending on value of `fileext`).
+#' @keywords internal
+#' @export
+#' @importFrom utils download.file write.csv
+prep_sedt_resource <- function(resource,
+                               coords = c("lon", "lat"),
+                               file = NULL,
+                               fileext = "csv",
+                               placement = "surface",
+                               ...,
+                               arg = caller_arg(resource),
```

Comment:
I also wasn't quite sure of the logic for adding `...` and setting `file = NULL` as an argument in this function. It seemed to me that we'd envision this as an internal function that's only called by other functions in the package, in which case it appears that the `file` argument is never set to something besides the default. Were you seeing a use case for users calling this function directly?

Reply:
See above for my thoughts on passing `...` to this function.

Giving file a `NULL` default value allows a user to set resource as a URL where a file is located and pass the filename to use for the downloaded file as a separate parameter. A user could pass a GeoPackage file as the resource, e.g. "parks.gpkg", and then user the file argument to set the path and file name of the converted data, e.g. "path_to_converted_file/sedtR_parks.csv". I can't recall if I included any documentation for this feature and it may not be essential.